### PR TITLE
Add Wizard and Layout.Wizard components

### PR DESCRIPTION
**Pull request by mairatma**
I've created two separate components: `Wizard` and `Layout.Wizard`.

`Wizard` is the one that contains the `Wizard.Footer` from a previous pr I've sent. It also has the logic for deciding which of the child step contents to show. It can receive all steps as its children, but it will only show the current one.

`Layout.Wizard`is an integration between `Layout` and `Wizard`. It can be used inside a `<Layout.Main>`, and it handles:
  * The layout header with the current step's name, automatically as the step changes.
  * The position of the wizard inside the layout.

In the end, this is the result for `Layout.Wizard`:
layout wizard

Note that the animation isn't smooth in the GIF, but it's quite smooth in reality :)

PS.: All new components are covered with tests.

### CODE REVIEW

File: stories/Layout/wizard.stories.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        <Dropdown item text="Maira Bello">
+          <Dropdown.Menu>
+            <Dropdown.Item text="Account" />
+            <Dropdown.Item text="Logout" />
+          </Dropdown.Menu>
+        </Dropdown>
+      </Menu.Menu>
+    </Layout.Topbar>
+    <Layout.Main>
+      <Layout.Wizard steps={['First Step', 'Second Step', 'Third Step']}>
```
Comment (mairatma) [inline stories/Layout/wizard.stories.js]: This is what you'd need to build a Wizard page. Basically, just use `Layout.Wizard` as your main content, passing an array with the step names, and `Wizard.Step` components for each step, in order. The Wizard will take care of the header, selecting the right `Wizard.Step`, the footer, and everything else.

File: components/Layout/Wizard/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+import { Layout } from '../'
+import Wizard from '../../Wizard'
```
Comment (kaueburiti) [inline components/Layout/Wizard/index.js]: are we using relative path as the default pattern?

File: components/Wizard/Footer/index.js [hunk 3]
```diff
@@ -55,18 +51,19 @@ class WizardFooter extends Component {
...
 
-  handleChangeStep = newStepIndex => {
-    this.setState({ currentStep: newStepIndex })
-    this.props.onChangeStep(newStepIndex)
+  renderControls = props => {
+    const { controls, steps } = this.props
+    return controls ? (
+      controls(props)
+    ) : (
+      <WizardFooterBasicControls {...props} totalSteps={steps.length} />
```
Comment (FranciscoGileno) [inline components/Wizard/Footer/index.js]: Great!

File: stories/Layout/wizard.stories.js (hunk 1 above)
Comment (kaueburiti) [reply stories/Layout/wizard.stories.js]: We can use the WIzard's component in an independent way, right? I mean, we can use it as a small section in an page, or can we just use it as the occupying whole page?

**Comment by FranciscoGileno** [comment]
Will `Layout.Wizard` always use `WizardFooterBasicControls` ?

This is a really awesome abstraction! Congratulations.

File: stories/Layout/wizard.stories.js (hunk 1 above)
Comment (mairatma) [reply stories/Layout/wizard.stories.js]: Yep, you can just use the `Wizard` component. It'll probably never happen, but I've separated them anyway. I've added a story just for the `Wizard`, btw.

**Pull request merged**